internal/commands/build.go

```diff
@@ -0,0 +1,259 @@
+		return nil, err
+	}
+
+	buildopts["invocation-image"] = build.Options{
+		Inputs:      build.Inputs{
+			InStream: buildContext,
+			ContextPath: "-",
+		},
+		Session:     []session.Attachable{authprovider.NewDockerAuthProvider(os.Stderr)},
+	}
+
+	d, err := driver.GetDriver(ctx, "buildx_buildkit_default", nil, dockerCli.Client(), nil, "", nil)
+	if err != nil {
+		return nil, err
+	}
+	driverInfo := []build.DriverInfo{
+		{
+			Name:   "default",
+			Driver: d,
+		},
+	}
+
+	ctx2, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	pw := progress.NewPrinter(ctx2, os.Stderr, opt.progress)
+	resp, err := build.Build(ctx2, driverInfo, buildopts, dockerAPI(dockerCli), dockerCli.ConfigFile(), pw)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println("Successfully built service images")
+	if logrus.IsLevelEnabled(logrus.DebugLevel)	{
+		dt, err := json.MarshalIndent(resp, "", "   ")
+		if err != nil {
+			return nil, err
+		}
+		logrus.Debug(string(dt))
+	}
+
+	for service, r := range resp {
+		digest := r.ExporterResponse["containerimage.digest"]
+		if service == "invocation-image" {
+			bundle.InvocationImages[0].Digest = digest
+		} else {
+			image := bundle.Images[service]
+			image.Image = fmt.Sprintf("%s:%s-%s", bundle.Name, bundle.Version, service)
+			image.ImageType = cnab.ImageTypeDocker
+			image.Digest = digest
+			bundle.Images[service] = image
+		}
+		fmt.Printf("    - %s : %s\n", service, digest)
+	}
+
+	if logrus.IsLevelEnabled(logrus.DebugLevel) {
+		dt, err := json.MarshalIndent(resp, "", "   ")
+		if err != nil {
+			return nil, err
+		}
+		logrus.Debug(string(dt))
+	}
+
+	var ref reference.Named
+	ref, err = getNamedTagged(opt.tag)
+	if err != nil {
+		return nil, err
+	}
+	if ref == nil {
+		b := bytes.Buffer{}
+		_, err := bundle.WriteTo(&b)
+		if err != nil {
+			return nil, err
+		}
+		digest := digest.SHA256.FromBytes(b.Bytes())
+		ref	= sha{digest}
+	}
+
+
+	if opt.out != "" {
+		b, err := json.MarshalIndent(bundle, "", "  ")
+		if err != nil {
+			return ref, err
+		}
+		if opt.out == "-" {
+			_, err = os.Stdout.Write(b)
+		} else {
+			err = ioutil.WriteFile(opt.out, b, 0644)
+		}
+		return ref, err
+	}
+
+	if err := persistInBundleStore(ref, bundle); err != nil {
+		return ref, err
+	}
+
+	return ref, nil
+}
+
+type sha struct {
+	d digest.Digest
+}
+
+func (s sha) Digest() digest.Digest {
+	return s.d
+}
+
+func (s sha) String() string {
+	return s.d.String()
+}
+
+func (s sha) Name() string {
+	return s.d.String()
+}
+
+var _ reference.Named = sha{""}
+var _ reference.Digested = sha{""}
+
+
+/// FIXME copy from vendor/github.com/docker/buildx/commands/util.go:318 could probably be made public
+func dockerAPI(dockerCli command.Cli) *api {
+	return &api{dockerCli: dockerCli}
+}
+
+type api struct {
+	dockerCli command.Cli
+}
+
+func (a *api) DockerAPI(name string) (dockerclient.APIClient, error) {
+	if name == "" {
+		name = a.dockerCli.CurrentContext()
```

Comment:
This doesn't really do anything right?

Reply:
indeed. Sounds weird this didn't made build to fail during testing. Does it mean this is actually unused by Buildx ?

Reply:
passing `nil` just works running local tests... need to get more infos from buildx team to understand the need for this parameter